org.wso2.carbon.consent.mgt.core/src/main/java/org/wso2/carbon/consent/mgt/core/dao/JdbcTemplate.java

```diff
@@ -0,0 +1,219 @@
+     * @param <T>
+     */
+    public <T extends Object> void executeUpdate(String query) throws DataAccessException {
+        try (Connection connection = dataSource.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            doInternalUpdate(null, preparedStatement);
+        } catch (SQLException e) {
+            logDebugInfo("Error in performing database update: {}", query);
+            throw new DataAccessException(ConfigurationConstants.ErrorMessages.ERROR_CODE_DATABASE_QUERY_PERFORMING
+                    .getMessage() + query, ConfigurationConstants.ErrorMessages.ERROR_CODE_DATABASE_QUERY_PERFORMING
+                    .getCode(), e);
+        }
+    }
+
+    /**
+     * Executes the jdbc insert/update query.
+     *
+     * @param query       The SQL for insert/update.
+     * @param queryFilter Query filter to prepared statement parameter binding.
+     * @param bean        the Domain object to be inserted/updated.
+     * @param <T>
+     */
+    public <T extends Object> int executeInsert(String query, QueryFilter queryFilter, T bean, boolean fetchInsertedId)
+            throws DataAccessException {
+        try (Connection connection = dataSource.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            doInternalUpdate(queryFilter, preparedStatement);
+            if (fetchInsertedId) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Mapping generated key (Auto Increment ID) to the object");
+                }
+                try (ResultSet generatedKeys = preparedStatement.getGeneratedKeys()) {
+                    if (generatedKeys.next()) {
+                        int resultId = generatedKeys.getInt(1);
+                        if (logger.isDebugEnabled()) {
+                            logger.debug("Newly inserted ID (Auto Increment ID) is {} for the bean {} ",
+                                    resultId, bean);
+                        }
+                        return resultId;
+                    } else {
+                        throw new SQLException(ConfigurationConstants.ErrorMessages
+                                .ERROR_CODE_AUTO_GENERATED_ID_FAILURE.getMessage());
+                    }
+                }
+            }
+        } catch (SQLException e) {
+            logDebugInfo("Error in performing database insert: {} with parameters {}", query, queryFilter);
+            throw new DataAccessException(ConfigurationConstants.ErrorMessages.ERROR_CODE_DATABASE_QUERY_PERFORMING.getMessage()
+                    + query, ConfigurationConstants.ErrorMessages.ERROR_CODE_DATABASE_QUERY_PERFORMING.getCode(), e);
+        }
+        return 0;
+    }
+
+    private <T extends Object> void doInternalUpdate(QueryFilter queryFilter, PreparedStatement preparedStatement)
+            throws SQLException, DataAccessException {
+        if (queryFilter != null) {
```

Comment:
Shall we add a new line after the method signature and shall we fix other places as well?